internal/git/git.go

```diff
@@ -173,6 +186,98 @@ func (g *GitOperations) GetShaTime(ctx context.Context, sha string) (v1.Time, er
+	//	return err
+	//}
+	//logger.V(4).Info("Pulled branch", "branch", environmentBranch)
+
+	_, stderr, err = g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext), "pull", "--progress", "origin", environmentNextBranch)
+	if err != nil {
+		logger.Error(err, "could not git pull", "gitError", stderr)
+		return err
+	}
+	logger.V(4).Info("Pulled branch", "branch", environmentNextBranch)
+
+	_, stderr, err = g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext), "push", "--progress", "origin", environmentBranch)
+	if err != nil {
+		logger.Error(err, "could not git pull", "gitError", stderr)
+		return err
+	}
+	logger.V(4).Info("Pushed branch", "branch", environmentBranch)
+
+	return nil
+}
+
+// IsPullRequestRequired will compare the environment branch with the next environment branch and return true if a PR is required.
+// The PR is required if the diff between the two branches contain edits to yaml files.
+func (g *GitOperations) IsPullRequestRequired(ctx context.Context, environmentBranch, environmentNextBranch string) (bool, error) {
+	logger := log.FromContext(ctx)
+
+	//environmentNextBranch := environmentBranch + "-next"
+
+	if g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext) == "" {
+		return false, fmt.Errorf("no repo path found")
+	}
+
+	// Checkout the environment branch
+	_, stderr, err := g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext), "checkout", "--progress", "-B", environmentBranch, fmt.Sprintf("origin/%s", environmentBranch))
+	if err != nil {
+		logger.Error(err, "could not git checkout", "gitError", stderr)
+		return false, err
+	}
+	logger.V(4).Info("Checked out branch", "branch", environmentBranch)
+
+	// Fetch the next environment branch
+	_, stderr, err = g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext), "fetch", "origin", environmentNextBranch)
+	if err != nil {
+		logger.Error(err, "could not fetch branch", "gitError", stderr)
+		return false, err
+	}
+	logger.V(4).Info("Fetched branch", "branch", environmentNextBranch)
+
+	// Get the diff between the two branches
+	stdout, stderr, err := g.runCmd(ctx, g.pathLookup.Get(g.gap.GetGitHttpsRepoUrl(*g.repoRef)+g.pathContext), "diff", fmt.Sprintf("origin/%s", environmentBranch), fmt.Sprintf("origin/%s", environmentNextBranch), "--name-only", "--diff-filter=ACMRT")
+	if err != nil {
+		logger.Error(err, "could not get diff", "gitError", stderr)
+		return false, err
+	}
+	logger.V(4).Info("Got diff", "diff", stdout)
+
+	// Check if the diff contains any YAML files if so we expect a manifest to have changed
+	// TODO: This is temporary check we should add some path globbing support to the specs
+	for _, file := range strings.Split(stdout, "\n") {
+		if strings.HasSuffix(file, ".yaml") || strings.HasSuffix(file, ".yml") {
+			logger.V(4).Info("YAML file changed", "file", file)
+			return true, nil
+		}
+	}
```

Comment:
Could probably replace this with two globs on the git command https://stackoverflow.com/a/8554987/684776